lib/ipam/ipam.go

```diff
@@ -291,7 +301,9 @@ func (c ipamClient) AssignIP(ctx context.Context, args AssignIPArgs) error {
 		if err != nil {
 			log.Warningf("Update failed on block %s", block.CIDR.String())
 			if args.HandleID != nil {
-				c.decrementHandle(ctx, *args.HandleID, blockCIDR, 1)
+				if err := c.decrementHandle(ctx, *args.HandleID, blockCIDR, 1); err != nil {
```

Comment:
We were ignoring this error before, need to think if that was right or not so leaving a comment to remind me to check.